File: python/gpt/p105.py
"""
p105.py
Create `secret.txt`, write the base64 string and set restrictive permissions (0600).
"""
import os

fname = 'secret.txt'
with open(fname, 'w', encoding='utf-8') as f:
    f.write('U2FsdGVkX1/R+WzJcxgvX/Iw==')
os.chmod(fname, 0o600)
print(fname)

File: c/gpt/p105.c
#include <fcntl.h>
#include <unistd.h>
#include <sys/stat.h>
#include <string.h>
#include <stdio.h>

/* p105.c
   Create secret.txt and write the given base64 string, then set restrictive perms (0600)
*/
int main(void) {
    const char *fname = "secret.txt";
    const char *msg = "U2FsdGVkX1/R+WzJcxgvX/Iw==";
    int fd = open(fname, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
    if (fd < 0) { perror("open"); return 1; }
    if (write(fd, msg, strlen(msg)) < 0) { perror("write"); close(fd); return 1; }
    close(fd);
    if (chmod(fname, S_IRUSR | S_IWUSR) != 0) perror("chmod");
    return 0;
}
